include/sys/vm_object.h

```diff
@@ -3,14 +3,23 @@
 
 #include <sys/vm.h>
 #include <sys/vm_pager.h>
+#include <sys/mutex.h>
+#include <sys/refcnt.h>
+
+/*! \brief Virtual memory object
+ *
+ * Field marking and corresponding locks:
+ * (a) atomic
+ * (@) vm_object::mtx
+ */
 
-/* At the moment assume object is owned by only one vm_map */
 typedef struct vm_object {
-  vm_pagelist_t list;
-  vm_pagetree_t tree;
-  size_t size;
-  size_t npages;
-  vm_pager_t *pager;
+  mtx_t mtx;
+  vm_pagelist_t list;   /* (@) List of pages */
+  vm_pagetree_t tree;   /* (@) Tree of pages */
+  size_t npages;        /* (@) Number of pages */
+  vm_pager_t *pager;    /* Pager type and page fault function for object */
+  refcnt_t ref_counter; /* (a) How many objects refer to this object? */
```

Comment:
@cahirwpz Should we consider adding prefix to every field in this structure? (like `vo_*` or `vmo_*`) We follow this rule in other structures, and it is easier to read code when we access fields with descriptive names. If we want to follow this rule here it might be better to change this until our structure gets much bigger.

Reply:
@franciscozdo Yes, but in another PR.

@bkjg Please put it on your TODO list.